code/ship/ship.cpp

```diff
@@ -9072,46 +9127,54 @@ static void ship_auto_repair_frame(int shipnum, float frametime)
 	sip = &Ship_info[sp->ship_info_index];
 	objp = &Objects[sp->objnum];
 
-	if (sp->flags[Ship::Ship_Flags::Dying]) // do not repair if already dead 
+	// do not repair if already dead 
+	if (sp->flags[Ship::Ship_Flags::Dying])
 		return;
 
-	//Repair the hull...or maybe unrepair?
-	if(sip->hull_repair_rate != 0.0f)
+	// Repair the hull...or maybe unrepair?
+	if (!fl_equal(sip->hull_repair_rate, 0.0f))
```

Comment:
This should remain as a strict equality, we're not looking for the result of numerical operations which were intended to bring it to 0 but may have slightly missed due to float imprecision, we are looking specifically for *assigned to 0* (by the initialization code), in which case it will be *exactly* 0. All other cases should be acted upon.

Reply:
Good to know, thanks!